lib/fluent/plugin/filter_prometheus.rb

```diff
@@ -22,9 +22,9 @@ def configure(conf)
       @metrics = Fluent::Plugin::Prometheus.parse_metrics_elements(conf, @registry, labels)
     end
 
-    def filter_stream(tag, es)
-      instrument(tag, es, @metrics)
-      es
+    def filter(tag, time, record)
+      instrument(tag, { time => record }, @metrics)
```

Comment:
For additional optimization, add `instrument_single(tag, time, record ,metrics)` version is better.